content_scripts/vimium_frontend.coffee

```diff
@@ -110,8 +123,19 @@ initializePreDomReady = ->
   settings.addEventListener("load", LinkHints.init.bind(LinkHints))
   settings.load()
 
+  # Install normal mode. This is at the bottom of both the mode stack and the handler stack, and is never
+  # deactivated.
+  new NormalMode()
```

Comment:
I don't agree with this one.  You activate a mode by creating a new instance of the relevant class.  The mode itself knows how to install itself.  And modes deactivate themselves when they're done.  Their logic is self contained.  They're self managing after launch.  This is the right way to handle things.  Code elsewhere does not need to worry about what mode we're in.